src/Stack/Build/Haddock.hs

```diff
@@ -209,16 +208,17 @@ generateHaddockIndex descr envOverride wc dumpPackages docRelFP destDir = do
                                 , srcInterfaceModTime
                                 , srcInterfaceAbsFile
                                 , destInterfaceAbsFile )
+    tryGetModificationTime :: FilePath -> IO (Either () UTCTime)
+    tryGetModificationTime = tryJust (guard . isDoesNotExistError) . getModificationTime
+    copyPkgDocs :: (a, UTCTime, Path Abs File, Path Abs File) -> IO ()
     copyPkgDocs (_,srcInterfaceModTime,srcInterfaceAbsFile,destInterfaceAbsFile) = do
-        -- | Copy dependencies' haddocks to documentation directory.  This way, relative @../$pkg-$ver@
+        -- Copy dependencies' haddocks to documentation directory.  This way, relative @../$pkg-$ver@
```

Comment:
In order to avoid problems like this one, would it be a good idea to let Travis build the haddocks too?